src/invidious/helpers/i18n.cr

```diff
@@ -78,7 +78,7 @@ def load_all_locales
   return locales
 end
 
-def translate(locale : String?, key : String, text : String | Nil = nil) : String
+def translate(locale : String?, key : String, text : String | Nil = nil, texts : Hash(String, String) | Nil = nil) : String
```

Comment:
You can do the following, and use `text.is_a?(Type)` to differentiate between the type of inputs:
```suggestion
def translate(locale : String?, key : String, text : String | Hash(String, String) | Nil = nil) : String
```